src/Symfony/Component/Yaml/Inline.php

```diff
@@ -466,14 +468,16 @@ private static function evaluateScalar($scalar, $references = array())
                         return '0' == $scalar[1] ? octdec($scalar) : (((string) $raw === (string) $cast) ? $cast : $raw);
                     case is_numeric($scalar):
                     case preg_match(self::getHexRegex(), $scalar):
+                        $scalar = str_replace('_', '', $scalar);
+
                         return '0x' === $scalar[0].$scalar[1] ? hexdec($scalar) : (float) $scalar;
                     case '.inf' === $scalarLower:
                     case '.nan' === $scalarLower:
                         return -log(0);
                     case '-.inf' === $scalarLower:
                         return log(0);
-                    case preg_match('/^(-|\+)?[0-9,]+(\.[0-9]+)?$/', $scalar):
```

Comment:
That tool is awesome, and yes, the `,` is not allowed as a separator (it renders as string when one is used in a number). I'd rather not touch it, as it would introduce a bc break.

But `123_` seems to be a valid integer too, so I need to keep that (it will simplify some regex)